### ISSUE TITLE
SSSD missing on first run CentOS - domain join fails

### ISSUE THREAD

**Issue by spacetourist** [walkamongus/realmd#57]
When performing first Puppet run on a minimal CentOS 1804 installation, joining realm using password fails due to missing dependencies. 

`
Notice: /Stage[main]/Realmd::Install/Package[sssd]/ensure: created                          
Info: Class[Realmd::Config]: Scheduling refresh of Class[Realmd::Join]     
Info: Class[Realmd::Join]: Scheduling refresh of Class[Realmd::Join::Password]                                                                                                 
Info: Class[Realmd::Join::Password]: Scheduling refresh of Exec[realm_join_with_password]                                                          
Notice: /Stage[main]/Realmd::Join::Password/File[/usr/libexec/realm_join_with_password]/ensure: defined content as '{md5}12341234123412341234123412341234'
Notice: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]/returns: Password for Administrator: See: journalctl REALMD_OPERATION=r796.54132
Notice: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]/returns: realm: Couldn't join realm: Necessary packages are not installed: oddjob, oddjob-mkhomedir, sssd, samba-common-tools
Error: '/usr/libexec/realm_join_with_password realm join production.network --unattended --user=Administrator --computer-name=FR-P-RTP-2' returned 1 instead of one of [0]
Error: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]/returns: change from 'notrun' to ['0'] failed: '/usr/libexec/realm_join_with_password realm join production.network --unattended --user=Administrator --compute
r-name=FR-P-RTP-2' returned 1 instead of one of [0]                                                                                                                     
Notice: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]/returns: Password for Administrator: See: journalctl REALMD_OPERATION=r796.54150
Notice: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]/returns: realm: Couldn't join realm: Necessary packages are not installed: oddjob, oddjob-mkhomedir, sssd, samba-common-tools
Error: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]: Failed to call refresh: '/usr/libexec/realm_join_with_password realm join production.network --unattended --user=Administrator --computer-name=FR-P-RTP-2' ret
urned 1 instead of one of [0]                                 
Error: /Stage[main]/Realmd::Join::Password/Exec[realm_join_with_password]: '/usr/libexec/realm_join_with_password realm join production.network --unattended --user=Administrator --computer-name=FR-P-RTP-2' returned 1 instead of one o
f [0]
`

**Comment by walkamongus** [comment]
The sssd package should be installed by https://github.com/walkamongus/realmd/blob/ecdeb7cada00c004bc85db4a1c04ed745e750adc/manifests/install.pp#L19, which references data at https://github.com/walkamongus/realmd/blob/ecdeb7cada00c004bc85db4a1c04ed745e750adc/data/common.yaml#L9.

The first line of the output indicates this occurred: `Notice: /Stage[main]/Realmd::Install/Package[sssd]/ensure: created` -- it seems like there is something else going on here besides the sssd package not being installed.

**Comment by spacetourist** [comment]
I agree, however I presumed it was a matter of this not being part of the ensure_packages() check during install. I might be wrong but I have been having mixed results on a number of server installations, sometimes we are there in time with SSSD, sometimes not. I hoped this would be solved through use of stdlib function as our means of installation.

https://github.com/walkamongus/realmd/blob/ca8a1d233c5ee9ce826c910e7bbbe85d0ac7a068/manifests/install.pp#L23

Having said that I am not so sure that this would make any difference to the Puppet behaviour. Do you know of any other way to make the package wait for SSSD installation to complete / have any other ideas?
